Fix the wrong Cypher commands for showing a specific index and editorial updates of the page (#1669)
neo4j/docs-operations: modules/ROOT/pages/performance/index-configuration.adoc

. Restart Neo4j.

+
The recommended way to set an index provider for an index is to use the `OPTIONS` clause for index creation, unique property constraint creation, and node key creation.
For more information, see link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/indexes-for-search-performance[Cypher Manual -> Indexes for search performance] and link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/constraints[Cypher manual -> Constraints].

[[index-configuration-btree-migration]]
=== Index migration

> When upgrading a 3.5 store to 4.4.34, all indexes are upgraded to the latest index version and rebuilt automatically, except for those previously using Lucene for single-property strings.

Author: If it is hard-coded, yes.

> When upgrading a 3.5 store to 4.4.34, all indexes are upgraded to the latest index version and rebuilt automatically, except for those previously using Lucene for single-property strings.

Author: That's a good catch. I don't know why I have changed it to 4.4.34.

. Restart Neo4j.

+
The recommended way to set an index provider for an index is to use the `OPTIONS` clause for index creation, unique property constraint creation, and node key creation.
For more information, see link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/indexes-for-search-performance[Cypher Manual -> Indexes for search performance] and link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/constraints[Cypher manual -> Constraints].

[[index-configuration-btree-migration]]
=== Index migration

> When upgrading a 3.5 store to 4.4.34, all indexes are upgraded to the latest index version and rebuilt automatically, except for those previously using Lucene for single-property strings.

Author: ```suggestion
When upgrading a 3.5 store to {neo4j-version-exact}, all indexes are upgraded to the latest index version and rebuilt automatically, except for those previously using Lucene for single-property strings.
```

For a complete list of Neo4j procedures, see link:{neo4j-docs-base-uri}/operations-manual/{page-version}/reference/procedures/[Operations Manual -> Procedures].

`dbms.index.fulltext.default_analyzer`::
The default analyzer for full-text indexes.
This setting takes effect when creating an index and is remembered as an index-specific setting.
+
The list of possible analyzers is available through the `db.index.fulltext.listAvailableAnalyzers()` Cypher procedure.
+
Unless otherwise specified, the default analyzer is standard-no-stop-words, the same as the StandardAnalyzer from Lucene, except no stop-words are filtered out.

`dbms.index.fulltext.eventually_consistent`::
Used to declare whether full-text indexes should be eventually consistent or not.
This setting only has effect when a full-text index is created and is remembered as an index-specific setting from then on.
+
Indexes usually are fully consistent, and the committing of a transaction returns once both the store and the indexes are updated.

> On the other hand, eventually_consistent full-text indexes are not updated as part of the commit; instead, their updates are queued up and applied in a background thread.

Author: ```suggestion
On the other hand, eventually consistent full-text indexes are not updated as part of the commit; instead, their updates are queued up and applied in a background thread.
```

This means there can be a short delay between committing a change and the change becoming visible via any eventually consistent full-text indexes.
This delay is just an artefact of the queueing and is usually relatively small since.
Eventually, consistent indexes are updated "as soon as possible".
+
This is turned off by default, and full-text indexes are fully consistent.

`dbms.index.fulltext.eventually_consistent_index_update_queue_max_length`::

> Eventually_consistent full-text indexes have their updates queued up and applied in a background thread, and this setting determines the maximum size of that update queue.

Author: ```suggestion
Eventually consistent full-text indexes have their updates queued up and applied in a background thread, and this setting determines the maximum size of that update queue.
```

If the maximum queue size is reached, the commit transactions block and wait until there is more room in the queue before adding more updates.
+

> This setting applies to all eventually_consistent full-text indexes, and they all use the same queue.

Author: ```suggestion
This setting applies to all eventually consistent full-text indexes, and they all use the same queue.
```

== Future indexes

Two new index types, range and point index, will be introduced in 5.0.
They will, together with the text index, replace the deprecated B-tree indexes.

> Like the B-tree index, the range index will index all types of values and be good for exact lookups on all types of values, range scans,

Reviewer: ```suggestion
Like the B-tree index, the range index will index all types of values and be good for exact lookups on all types of values, range scans, full scans, and prefix searches.
```

> full scans, and prefix searches.

Reviewer: ```suggestion
```
